aggregations/db_tables/daily_add_key_by_app_count.py

```diff
@@ -0,0 +1,94 @@
+import datetime
+
+from . import DAY_LEN_SECONDS, daily_start_of_range, time_json
+from ..periodic_aggregations import PeriodicAggregations
+
+
+class DailyAddKeyByAppCount(PeriodicAggregations):
+    def dependencies(self) -> list:
+        return ["near_ecosystem_entities"]
+
+    @property
+    def sql_create_table(self):
+        return """
+            CREATE TABLE IF NOT EXISTS daily_add_key_by_app_count
+            (
+                collected_for_day  DATE NOT NULL,
+                app_id TEXT NOT NULL,
+                receipt_receiver_account_id TEXT NOT NULL,
+                add_key_count INTEGER NOT NULL,
+                PRIMARY KEY (collected_for_day, app_id, receipt_receiver_account_id )
+            );
+            CREATE INDEX IF NOT EXISTS daily_add_key_by_app_count_idx
+                ON daily_add_key_by_app_count (collected_for_day, add_key_count DESC)
+        """
+
+    @property
+    def sql_drop_table(self):
+        return """
+            DROP TABLE IF EXISTS daily_add_key_by_app_count
+        """
+
+    @property
+    def sql_select(self):
+        all_apps = """
+            SELECT token as contract_id, 
+            slug as app_id
+            FROM public.near_ecosystem_entities e, unnest(string_to_array(e.contract, ', ')) s(token)
+            WHERE app = TRUE AND length(contract)>0
+        """
+
+        with self.analytics_connection.cursor() as analytics_cursor:
+            analytics_cursor.execute(all_apps)
+            app_contracts = analytics_cursor.fetchall()
+
+        string = ""
+        for index, tuple in enumerate(app_contracts):
+            contract = tuple[0]
+            app = tuple[1]
+            string += (
+                "WHEN args ->'access_key' -> 'permission' -> 'permission_details' ->> 'receiver_id' LIKE '"
+                + contract.strip()
+                + "' THEN '"
+                + app.strip()
+                + "'"
+            )
+
+        opener = """SELECT CASE """
+        closer = """ELSE 'All Others' END as app_id ,
+        receipt_receiver_account_id,
+        count(*) as new_accounts_count
+        FROM public.action_receipt_actions a
+        WHERE action_kind IN ('ADD_KEY')
+            AND args ->'access_key' -> 'permission' ->> 'permission_kind' = 'FUNCTION_CALL'
+            AND args ->'access_key' -> 'permission' -> 'permission_details' ->> 'receiver_id' != receipt_receiver_account_id 
+            AND args ->'access_key' -> 'permission' -> 'permission_details' ->> 'receiver_id' != 'near'
+            AND receipt_included_in_block_timestamp  >= %(from_timestamp)s
+            AND receipt_included_in_block_timestamp  < %(to_timestamp)s
+        GROUP BY 1,2
```

Comment:
Have you tried to compose this query manually and measure the performance?

Reply:
Yes, I did extensive testing.  This was the quickest solution I found, even with the dependency.

Reply:
Could you please share the details of the testing? And the query that you've composed